### ISSUE TITLE
Add an animation node that will mirror animation on a skeleton.

### ISSUE THREAD

**Issue by RonYanDaik** [godotengine/godot-proposals#8007]
### Describe the project you are working on

A 3D game with fighting animations.

### Describe the problem or limitation you are having in your project

I would like to use one punching animation for both hands but don't want to create 2x copies of same animation but for opposite part.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

An animation node that you can add to the animation tree.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Adding a node that will seek through skeleton bones with same name and suffix _L _R and swap animation data between mirrored bones.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I couldn't find a way to achieve this with script. (Maybe I could swap animation tracks names? I'm not sure.)

### Is there a reason why this should be core and not an add-on in the asset library?

It additional node that improves quality of life.

**Comment by Mickeon** [comment]
This sounds very situational and it could probably be done with an addon with the right knowledge.

**Comment by RonYanDaik** [comment]
> This sounds very situational 

Idk. Every project I've been working on I thought that something like this would be very very handy.

**Comment by fire** [comment]
I think this is possible if you're ok with using the godot humanoid profile. Are you ok with that?

**Comment by Illauriel** [comment]
Chiming in, because I need this feature in my project too, and wanted to write a similar proposal.

> I think creating a feature enhancement for this is possible if you're ok with using the godot humanoid profile. Are you ok with that?

I'd actually love that! Not necessarily as an animation node, but maybe as a flag somewhere in the BoneMap, as it only relates to laterally symmetrical armatures.

I could think of at least a few more use-cases in context of animated 3d games:
- Ambidextry and Dual-weilding weapons - While some animation sets available on the market have "Twin x" subsets, in some gameplay setups you would want to use right-handed attack animations for the left side too. (As is my case)
- Left-handed characters - Sometimes, especially in RPGs user might want to create left-handed characters, but hand-crafting and then managing double the amount of animations just for role-playing purposes would be a huge time expenditure.
- Injury reactions and other misc animations - Things like walking with a limp or wincing at the hit to different sides of the body are some examples, but many other misc animations might benefit from this (like playing a "wave" emote while holding an item in right hand.)

As for "why not have it as an add-on", an argument like "Unity and Unreal have it out of the box" might not be the best, but they do. Coming from Unity background, I've spent considerable time looking for a way to do this in Godot without resorting to hacks or "just making flipped copies of everything", and turned out it just doesn't exist for now.

TL;DR: It's a pretty rare use case, but if it comes up, this enhancement would save users a ton of effort.

**Comment by lyuma** [comment]
@jmarceno I'm not sure the approach of negating X scale was intended to be a serious suggestion or not, but it brings a particular comic to mind... (http://www.mspaintadventures.com/newindex.php?s=6&p=003200)

For some *specific* games with perfectly symmetrical art, it *could* work as a workaround, but sprite flipping, as it were, is simply not acceptable as a general purpose solution for this problem.

Personally I agree completely with @Illauriel that this deserves to be solved at the engine level. At a minimum, there should be a reasonable workflow/pipeline for creating such mirrored animation that doesn't involve manually flipping animations or assets. Personally, I can see this being helpful for creating walk cycles such as sideways motion without needing to create double the number of animations for left and right.

**Comment by dfendt** [comment]
This is a great addition! I work as a gameplay animator and the amount of work and assets we're able to cut by simply having a mirror node is huge. Start move, stop move, directional change, leans, attacks, climbing, jumping, falling, prop interaction, idle breakers, etc., plus anything that requires left and right foot versions (locomotion wise, its more or less x2 to most things). We use it everywhere because it solves multiple issues like less bloated systems, decrease the graphs footprint and project size, but it mainly halves the animator's work on affected features. Essentially, do you want to iterate on two animations or one? When you scale that to thousands of animations? No, a mirror node will do just fine :)

**Comment by Radivarig** [comment]
I would appreciate this being an option inside of AnimationNodeAnimation similar to how `play_mode` has Forward/Backward, to it also having `flip` or `mirror` True/False.

This is something that has existed in Unity since forever and is extremely useful.
<img width="283" height="493" alt="Image" src="https://github.com/user-attachments/assets/ed42ae04-f843-4c38-95db-f86729d00a93" />
